test/middlewares/request_options_test.exs

```diff
@@ -0,0 +1,18 @@
+defmodule Middlewares.OptionsTest do
+  use ExUnit.Case, async: false
+
+  test "Options are added to request settings" do
+    req = Crawly.Request.new("http://example.com")
+    middlewares = [
+      {
+        Crawly.Middlewares.RequestOptions,
+        [timeout: 30_000, recv_timeout: 15000]
+      }
+    ]
+    state = %{spider_name: :test_spider}
```

Comment:
Is the spider_name key relevant for the test? Could just use an empty map if there are no relevant state assertion

Reply:
yes, we don't need any special state as I see